packages/netlify-cms-core/src/lib/stringTemplate.js

```diff
@@ -61,12 +67,8 @@ export function compileStringTemplate(template, date, identifier = '', data = Ma
   }
 }
 
-function extractVars(template, regexp) {
-  const result = regexp.exec(template);
-  return result && [result[1]].concat(extractVars(template, regexp)).filter(v => v);
-}
-
 export function extractTemplateVars(template) {
   const regexp = RegExp(templateVariablePattern, 'g');
-  return extractVars(template, regexp);
+  const contentRegexp = RegExp(templateContentPattern, 'g');
+  return template.match(regexp).map(elem => elem.match(contentRegexp)[0]);
```

Comment:
Brilliant!

Reply:
yeah, but I forgot to handle the case where there are no matches and the first `match`-call returns `null` instead of `[]`